store/store.go

```diff
@@ -253,6 +253,20 @@ func useStaging() bool {
 	return osutil.GetenvBool("SNAPPY_USE_STAGING_STORE")
 }
 
+// Extend a base URL with additional unescaped paths.  (url.Parse handles
+// resolving relative links, which isn't quite what we want: that goes wrong if
+// the base URL doesn't end with a slash.)
+func urlJoin(base *url.URL, paths ...string) *url.URL {
+	if len(paths) == 0 {
+		return base
+	}
+	url := *base
+	url.RawQuery = ""
```

Comment:
So ... after some chat I think we agreed there are 2 bits to this conversation:

1. it's correct to clear the query when the path changes, but it's probably just noise for the store's use case
2. if the env var url contains a query then we should fail loudly (in apiURL or init perhaps?) because it's almost certainly not what the user intended and it's about to get ignore anyway